packages/backend/src/routes/records.ts

```diff
@@ -149,15 +155,28 @@ export function getRecordsRouter(orm: EntityManager) {
       return res.status(404).send();
     }
 
-    const execution = record.newExecution();
-    await orm.save(execution);
+    void runExecution(orm, record);
 
-    void Crawler.crawl(execution); // don't wait for the crawling to finish before sending the response
-
-    const response: ResponseType<"/records/{recordId}/run", "post"> =
-      execution.serialize();
-    return res.status(200).json(response);
+    return res.status(200);
   });
 
   return router;
 }
+
+async function runExecution(orm: EntityManager, record: WebsiteRecord) {
+  const execution = record.newExecution();
+  await orm.save(execution);
+
+  void Crawler.crawl(execution).then(async () => {
+    await setTimeout(execution.record.periodicity * 1000);
+
+    // ensuring that we don't keep executing if the record is deleted
+    const recordInDatabase = await orm.findOneBy(WebsiteRecord, {
+      id: execution.record.id,
+    });
+
+    if (recordInDatabase) {
+      void runExecution(orm, record);
+    }
+  });
```

Comment:
This looks good to me. Perhaps we could utilize the `setInterval` function in JS (see the suggestion below), but it's just the matter of readability for me, no hard blocker (I feel like this "recursive" approach might cause stack overflow at some point... very far away in the future. Not too sure though, with the use of the `void` operator).

```suggestion
  await Crawler.crawl(execution);
  
  const interval = setInterval(async () => {
    const recordInDatabase = await orm.findOneBy(WebsiteRecord, {
      id: execution.record.id,
    });

    if (!recordInDatabase) {
      return clearInterval(interval);
    }
    
    await Crawler.crawl(execution);
  });
```

Reply:
I switched it to use setInterval. I just kept it as void instead of await, since we don't need to wait for the crawl to complete before counting the time to the next one.